Sources/AWSLambdaRuntimeCore/ControlPlaneRequestEncoder.swift

```diff
@@ -0,0 +1,126 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+struct ControlPlaneRequestEncoder: _EmittingChannelHandler {
+    typealias OutboundOut = ByteBuffer
+
+    private var host: String
+    private var byteBuffer: ByteBuffer!
+
+    init(host: String) {
+        self.host = host
+    }
+
+    mutating func writeRequest(_ request: ControlPlaneRequest, context: ChannelHandlerContext, promise: EventLoopPromise<Void>?) {
+        self.byteBuffer.clear(minimumCapacity: self.byteBuffer.storageCapacity)
+
+        switch request {
+        case .next:
+            self.byteBuffer.writeString(.nextInvocationRequestLine)
+            self.byteBuffer.writeHostHeader(host: self.host)
+            self.byteBuffer.writeString(.userAgentHeader)
+            self.byteBuffer.writeString(.CRLF) // end of head
+            context.write(self.wrapOutboundOut(self.byteBuffer), promise: promise)
+            context.flush()
+
+        case .invocationResponse(let requestID, let payload):
+            let contentLength = payload?.readableBytes ?? 0
+            self.byteBuffer.writeInvocationResultRequestLine(requestID)
+            self.byteBuffer.writeHostHeader(host: self.host)
+            self.byteBuffer.writeString(.userAgentHeader)
+            self.byteBuffer.writeContentLengthHeader(length: contentLength)
+            self.byteBuffer.writeString(.CRLF) // end of head
+            if contentLength > 0 {
```

Comment:
`if let payload = payload && contentLength > 0` would be nice so we can drop the force unwrap

Reply:
Fixed.